hw/vendor/patches/pulp_riscv_dbg/0001-Use-lowrisc-instead-of-PULP-primitives.patch

```diff
@@ -85,81 +98,189 @@ index f742a2b..373f481 100644
+-    .dst_rst_ni  ( rst_ni               ),
+-    .dst_clear_i ( 1'b0                 ), // No functional reset from core side
+-                                           // used (only async).
+-    .dst_clear_pending_o( core_clear_pending ), // use the clear pending signal
+-                                                // to synchronously clear the
+-                                                // response FIFO in the dm_top
+-                                                // csrs
+-    .dst_clk_i   ( clk_i                ),
+-    .dst_data_o  ( core_dmi_req_o       ),
+-    .dst_valid_o ( core_dmi_valid_o     ),
+-    .dst_ready_i ( core_dmi_ready_i     )
++
 +  // TODO: Make it clean for synthesis.
++  logic jtag_combined_rstn;
++  always_ff @(posedge tck_i or negedge trst_ni) begin
++    if (!trst_ni) begin
++      jtag_combined_rstn <= '0;
++    end else if (jtag_dmi_cdc_clear_i) begin
++      jtag_combined_rstn <= '0;
++    end else begin
++      jtag_combined_rstn <= 1'b1;
++    end
++  end
++
++  logic combined_rstn_premux;
++  prim_flop_2sync #(
++    .Width(1),
++    .ResetValue(0)
++  ) u_combined_rstn_sync (
++    .clk_i,
++    .rst_ni(rst_ni & jtag_combined_rstn),
++    .d_i(1'b1),
++    .q_o(combined_rstn_premux)
+   );
+ 
+-  cdc_2phase_clearable #(.T(dm::dmi_resp_t)) i_cdc_resp (
+-    .src_rst_ni  ( rst_ni               ),
+-    .src_clear_i ( 1'b0                 ), // No functional reset from core side
+-                                           // used (only async ).
+-    .src_clear_pending_o(), // Not used
+-    .src_clk_i   ( clk_i                ),
+-    .src_data_i  ( core_dmi_resp_i      ),
+-    .src_valid_i ( core_dmi_valid_i     ),
+-    .src_ready_o ( core_dmi_ready_o     ),
+-
+-    .dst_rst_ni  ( trst_ni              ),
+-    .dst_clear_i ( jtag_dmi_cdc_clear_i ),
+-    .dst_clear_pending_o(), //Not used
+-    .dst_clk_i   ( tck_i                ),
+-    .dst_data_o  ( jtag_dmi_resp_o      ),
+-    .dst_valid_o ( jtag_dmi_valid_o     ),
+-    .dst_ready_i ( jtag_dmi_ready_i     )
++  logic combined_rstn;
++  prim_clock_mux2 #(
++    .NoFpgaBufG(1'b1)
++  ) u_rst_mux (
++    .clk0_i(combined_rstn_premux),
++    .clk1_i(test_rst_ni),
++    .sel_i(testmode_i),
++    .clk_o(combined_rstn)
+   );
+ 
+-  // We need to flush the DMI response FIFO in DM top using the core clock
+-  // synchronous clear signal core_dmi_rst_no. We repurpose the clear
+-  // pending signal in the core clock domain by generating a 1 cycle pulse from
+-  // it.
  
--    .dst_rst_ni  ( rst_ni           ),
--    .dst_clk_i   ( clk_i            ),
--    .dst_data_o  ( core_dmi_req_o   ),
--    .dst_valid_o ( core_dmi_valid_o ),
--    .dst_ready_i ( core_dmi_ready_i )
+-  logic                    core_clear_pending_q;
+-  logic                    core_dmi_rst_nq;
+-  logic                    clear_pending_rise_edge_detect;
 +  prim_fifo_async #(
 +    .Width       ( $bits(dm::dmi_req_t) ),
-+    .Depth       ( 4 )
++    .Depth       ( 1 )
```

Comment:
Does Depth 1 work? I thought it should be greater than or equal to 2 and should be power of 2.

Reply:
so i was looking [here](https://github.com/lowRISC/opentitan/blob/master/hw/vendor/pulp_riscv_dbg/src/dmi_jtag.sv#L117) , and it looked like to me the `dmi` fsm can only handle one at a time.  They issue a request and wait on the response before accepting another transaction. So it didn't look like it could ever accept more than 1.  

I changed it to 1 here also to see if anything would break.  Let me know what you think.